src/ansys/fluent/core/launcher/launcher.py

```diff
@@ -558,6 +559,364 @@ def __init__(self, launch_string):
+                )
+            else:
+                raise UnexpectedKeywordArgument(
+                    f"launch_fluent() got an unexpected keyword argument {next(iter(self.kwargs))}"
+                )
+        del self.kwargs
+
+        if self.additional_arguments is None:
+            self.argvals["additional_arguments"] = ""
+
+        if self.fluent_launch_mode == LaunchMode.PIM and self.start_watchdog:
+            logger.warning(
+                "'start_watchdog' argument for 'launch_fluent' is currently not supported "
+                "when starting a remote Fluent PyPIM client."
+            )
+
+        if self.dry_run and self.fluent_launch_mode != LaunchMode.CONTAINER:
+            logger.warning(
+                "'dry_run' argument for 'launch_fluent' currently is only "
+                "supported when starting containers."
+            )
+
+        if self.fluent_launch_mode != LaunchMode.STANDALONE:
+            arg_names = [
+                "env",
+                "cwd",
+                "topy",
+                "case_file_name",
+                "lightweight_mode",
+                "journal_file_names",
+                "case_data_file_name",
+            ]
+            invalid_arg_names = list(
+                filter(lambda arg_name: argvals[arg_name] is not None, arg_names)
+            )
+            if len(invalid_arg_names) != 0:
+                invalid_str_names = ", ".join(invalid_arg_names)
+                logger.warning(
+                    f"These specified arguments are only supported when starting "
+                    f"local standalone Fluent clients: {invalid_str_names}."
+                )
+
+
+class StandaloneLauncher(Launcher):
+    """Instantiates Fluent session in standalone mode."""
+
+    def __init__(
+        self,
+        new_session: Optional[Any] = None,
+        argvals: Optional[Any] = None,
+        lightweight_mode: Optional[Any] = None,
+        additional_arguments: Optional[Any] = None,
+        meshing_mode: Optional[Any] = None,
+        show_gui: Optional[Any] = None,
+        env: Optional[Any] = None,
+        cwd: Optional[Any] = None,
+        journal_file_names: Optional[Any] = None,
+        topy: Optional[Any] = None,
+        start_timeout: Optional[Any] = None,
+        cleanup_on_exit: Optional[Any] = None,
+        start_transcript: Optional[Any] = None,
+        start_watchdog: Optional[Any] = None,
+        case_file_name: Optional[Any] = None,
+        case_data_file_name: Optional[Any] = None,
+        **kwargs,
+    ):
+        for arg_name, arg_values in argvals.items():
+            setattr(self, arg_name, arg_values)
+        self.argvals = argvals
+        super(StandaloneLauncher, self).__init__(self.argvals)
+
+    def __call__(self, **argvals):
+        """Launch Fluent session in standalone mode."""
+        for arg_name, arg_values in argvals.items():
+            setattr(self, arg_name, arg_values)
+        self.argvals.update(argvals)
```

Comment:
Are the base class constraints being upheld? Given that you are currently implementing them directly in the base constructor, that seems unlikely.

Reply:
We will include this change in next PR. Thank you.